### PR TITLE
Better option parsing

### PR DESCRIPTION
**Pull request by tkralphs**
OK, I implemented improved option parsing based on my new-found knowledge of the `shift` command. We could get even fancier and go for use of `getopt`, this option parsing does not really error check much. But it seems to work, and it implements what you asked for in #2 and #1. It's easy to add short forms now, too. Maybe I'll do that quickly.

### CODE REVIEW

**Comment by svigerske** [comment]
Using a library like `getopt` would be nice. I don't know whether this introduces a dependency or can be included into the coinbrew script. Rather something for a different issue.

**Comment by tkralphs** [comment]
I looked at using `getopt` or `getopts`, but `getopts` (bash built-in) doesn't support short forms and `getopt` is an external binary that may not be on all systems. The value added is relatively small and it makes the whole thing a bit less transparent. I just added some fixes, including making it backward compatible to the old style in case documentation is out of date anywhere.

File: coinbrew [hunk 1]
```diff
@@ -253,8 +266,16 @@ function parse_args {
                 uninstall=true
                 ;;
             *)
-                echo "Unrecognized command...exiting"
-                exit 3
+                if [[ "$arg" == *=* ]] || [[ "$arg" == --* ]]; then
+                    configure_options["$arg"]=""
+                else
+                    if [[ $arg == *:* ]]; then
```
Comment (svigerske) [inline coinbrew]: Use `elif` ?

File: coinbrew [hunk 2]
```diff
@@ -253,8 +266,16 @@ function parse_args {
...
+                        main_proj=`echo $arg | cut -d ':' -f 1`
+                        main_proj_version=`echo $arg | cut -d ':' -f 2`
```
Comment (svigerske) [inline coinbrew]: `main_proj=${arg%%:*}`
`main_proj_version=${arg#*:}`
?
Like you did at the top for case `*=*`

**Comment by tkralphs** [comment]
OK, guess it's tested well enough then :).

**Pull request merged**